### ISSUE TITLE
BBVA Spain | Online Banking

### ISSUE THREAD

**Issue by pachulo** [PrivSec-dev/banking-apps-compat-report#213]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### App name

BBVA Spain | Online Banking

### Link to app

https://play.google.com/store/apps/details?id=com.bbva.bbvacontigo

### App version

12.10.30

### Country of the app

Spain

### Build Number

TQ1A.230205.002.2023020600

### Device list

Pixel 6a

### Profile app tested in

Owner profile

### Google Play installed

Installed

### Google Play services Network permission revoked?

- [ ] Revoked
- [X] Not revoked
- [ ] I did not have Google Play services installed

### Native code debugging

- [X] Enabled
- [ ] Disabled

### Exploit protection compatibility mode

- [ ] Enabled
- [X] Disabled

### Stock OS compatibility

- [X] Works
- [ ] Does not work
- [ ] Not tested

### Description of the app's functionality

Everything seems to work OK, even biometric authentication & signature of operations.

**Comment by hoovad** [comment]
Does BBVA Pay work?

**Comment by rolandchl** [comment]
Does BBVA Pay work?

**Comment by spring-onion** [comment]
@rolandchl Probably. Last report was just over a year ago.

**Comment by platix** [comment]
BBVA Pay does not work.
There seems to be some limitation in the OS that blocks the checks they perform and does not finish the process of adding the card. No checks are done using the google integrity API. It must be some other restrictions, maybe it is in the SELinux configuration.

04-01 23:48:59.217 15868 16288 W libc    : Access denied finding property "ro.debuggable"
04-01 23:48:59.213 15868 15868 W Thread-40: type=1400 audit(0.0:745054): avc:  denied  { read } for  name="u:object_r:userdebug_or_eng_prop:s0" dev="tmpfs" ino=811 scontext=u:r:untrusted_app:s0:c193,c256,c512,c768 tcontext=u:object_r:userdebug_or_eng_prop:s0 tclass=file permissive=0 app=com.bbva.bbvacontigo

Any suggestions?

**Comment by hoovad** [comment]
Indeed, i have tried switching to GrapheneOS and tried adding the card to BBVA, but it gave me a vague error like "This is not available at this time."

**Comment by rolandchl** [comment]
Indeed, Adding my card to BBVA Pay does not work. "Setting up the app for BBVA Pay" shows up for a few seconds but then it errors out with a vague error like "This is not available at this time". However, BBVA has used the Play Integrity API on first startup, and it is weird that a "This phone is rooted" error doesn't show up if it wont let me add a card.

**Comment by spring-onion** [comment]
Ah you mean a subfunctionality of the app, my bad I missed that, ignore my previous comment.

**Comment by rolandchl** [comment]
The problem the ePA App is having is unrelated to ours, i have created a new issue for this problem https://github.com/GrapheneOS/os-issue-tracker/issues/5201

**Comment by rolandchl** [comment]
Ok so upstream GrapheneOS should not be the problem. @platix Do you have any more relevant logs that can help us?

**Comment by platix** [comment]
I have compared the logs of a pixel device with graphene and one without, and that error message does not show up. There are many others on both devices but not that one.

**Comment by rolandchl** [comment]
Then it must be GrapheneOS that is the problem, they are wrong

**Comment by torturado** [comment]
BBVA Pay can no longer be configured on GrapheneOS.
The onboarding flow now stops after the “Setting up BBVA Pay” screen and returns the generic error “This service is not available at this time.”
Tested on Pixel 8 Pro (GrapheneOS 2025-04-05, Play Services installed)

**Comment by rolandchl** [comment]
> BBVA Pay can no longer be configured on GrapheneOS.
> The onboarding flow now stops after the “Setting up BBVA Pay” screen and returns the generic error “This service is not available at this time.”
> Tested on Pixel 8 Pro (GrapheneOS 2025-04-25, Play Services installed)

I have opened an issue on the GrapheneOS issue tracker but the issue was closed as the maintainer thought it was normal, i'm going to try opening a new issue.

**Comment by rolandchl** [comment]
Unless we cant get more logs or more problems that cause this problem, we can't pinpoint the issue

**Comment by torturado** [comment]
> > BBVA Pay can no longer be configured on GrapheneOS.
> > The onboarding flow now stops after the “Setting up BBVA Pay” screen and returns the generic error “This service is not available at this time.”
> > Tested on Pixel 8 Pro (GrapheneOS 2025-04-25, Play Services installed)
> 
> I have opened an issue on the GrapheneOS issue tracker but the issue was closed as the maintainer thought it was normal, i'm going to try opening a new issue.

yes, I saw it. But I couldn't get any logs from the app when the bbva pay error happens. I already emailed BBVA Support explaining what's happening with play integrity

**Comment by ainaracatgirl** [comment]
Can confirm this is happening to me as well on Android 16 (BP2A.250805.005) and BBVA Spain version 14.3.24. Even with exploit compatibility mode enabled.

**Comment by TonimatasDEV** [comment]
> Potential fix (as I haven't tried to use tap to pay with secure app spawning enabled):
> 
> * Disable secure app spawning
> * Go to the BBVA Spain app
> * Turn BBVA Pay on
> * Enable secure app spawning again
> 
> This worked for me.

That doesn't fix the problem to me. Service temporally unavailable.

**Comment by torturado** [comment]
> > Potential fix (as I haven't tried to use tap to pay with secure app spawning enabled):
> > 
> > * Disable secure app spawning
> > * Go to the BBVA Spain app
> > * Turn BBVA Pay on
> > * Enable secure app spawning again
> > 
> > This worked for me.
> 
> That doesn't fix the problem to me. Service temporally unavailable.

I tried it and it doesn't fix it for me either.

**Comment by us77ipis** [comment]
For me it doesn't work either.

**Comment by TonimatasDEV** [comment]
@us77ipis @torturado With the latest updates of BBVA and GrapheneOS now I can pay with my phone!

**Comment by us77ipis** [comment]
@TonimatasDEV I just updated everything and for me still the same happens when trying to enable BBVA Pay. What are your settings? What exactly did you do to get it working? Thanks.

**Comment by us77ipis** [comment]
@TonimatasDEV My BBVA app version is v14.6.11.251226 and build v14.6.42.260122 and Play Store says no updates available.

**Comment by ainaracatgirl** [comment]
Disable secure app spawning and keep it disabled. It should work. Ideally this would be a per-app toggle, but we can't do much. The newest GOS update is meant to also fix it by copying Android's zygote spawning.

**Comment by torturado** [comment]
<img width="1008" height="2244" alt="Image" src="https://github.com/user-attachments/assets/ee0aa880-cb69-4632-9658-4464e21c4085" /> yeah guys i disabled secure app spawning and it worked the BBVA pay option

**Comment by us77ipis** [comment]
And after successfully enabling BBVA Pay, can you then again enable secure app spawning and keep paying or will that break and it must be kept disabled?

**Comment by us77ipis** [comment]
I have now the latest version of BBVA app and also updated GrapheneOS and disabled secure app spawning. I still cannot activate BBVA Pay, it always gives a toast message saying "Service temporarily not available". Did you change any other settings? Maybe it has something to do with Google Play services permissions? @torturado @TonimatasDEV @ainaracatgirl

**Comment by torturado** [comment]
@us77ipis I don't have google play services activated. I've only disabled secure app spawning and it worked.

**Comment by rolandchl** [comment]
Can i get a quick status report? I would like to know if it still works

**Comment by us77ipis** [comment]
I have not been able to get it working yet.

**Comment by rolandchl** [comment]
@torturado @ainaracatgirl

**Comment by ainaracatgirl** [comment]
I no longer use GrapheneOS, disabling secure app spawning and keeping it disabled worked for me without issue. Please stop mentioning me.

**Comment by torturado** [comment]
> [@torturado](https://github.com/torturado) [@ainaracatgirl](https://github.com/ainaracatgirl)

i got it working and I tried paying with BBVA pay and it worked using what ainara said: disable secure app spawning

**Comment by us77ipis** [comment]
With BBVA App version 14.8.9 it works for me now on my Pixel 9a without disabling any security mechanisms of GrapheneOS, no disabling of secure app spawning necessary anymore!

**Comment by torturado** [comment]
> With BBVA App version 14.8.9 it works for me now on my Pixel 9a without disabling any security mechanisms of GrapheneOS, no disabling of secure app spawning necessary anymore!

thx, i've been able to add it without disabling secure app spawning with BBVA pay

**Comment by platix** [comment]
It works for me too; the problem seems to have been completely resolved

**Comment by rolandchl** [comment]
I can confirm that BBVA Spain works on GrapheneOS, all functionality so
far, even NFC. Without Google Play Services even!
